runtime/interpreter/interpreter.go

```diff
@@ -978,6 +980,47 @@ func (interpreter *Interpreter) declareAttachmentValue(
 	return interpreter.declareCompositeValue(declaration, lexicalScope)
 }
 
+// evaluates all the implicit default arguments to the default destroy event
+//
+// the handling of default arguments makes a number of assumptions to simplify the implementation;
+// namely that a) all default arguments are lazily evaluated at the site of the invocation,
+// b) that either all the parameters or none of the parameters of a function have default arguments,
+// and c) functions cannot currently be explicitly invoked if they have default arguments
+//
+// if we plan to generalize this further, we will need to relax those assumptions
+func (interpreter *Interpreter) evaluateDefaultDestroyEvent(
+	containerComposite *CompositeValue,
+	compositeDecl *ast.CompositeDeclaration,
+	compositeType *sema.CompositeType,
```

Comment:
It is not quite clear what `compositeType` is here – is it the composite type of the `ResourceDestroyed` event?

Reply:
Removed this argument, it was unnecessary